src/lib/dialog/dialog-container.ts

```diff
@@ -77,7 +80,8 @@ export class MdDialogContainer extends BasePortalHost implements OnDestroy {
     private _ngZone: NgZone,
     private _renderer: Renderer,
     private _elementRef: ElementRef,
-    private _focusTrapFactory: FocusTrapFactory) {
+    private _focusTrapFactory: FocusTrapFactory,
+    @Optional() @Inject(DOCUMENT) private _document: any) {
```

Comment:
Chatted with Chuck and this is a known issue with the metadata extractor (https://github.com/angular/angular/issues/15424). Until that's fixed, we can inject it as `any`   without the `private` sugar and then manually assign it to the field with type `Document`.